lib/sdotypes.c

```diff
@@ -2082,33 +2110,11 @@ void sdo_public_key_write(sdow_t *sdow, sdo_public_key_t *pk)
 	if (!sdow)
 		return;
 
-	if (pk == NULL || pk->key1->byte_sz == 0) {
-		// Write null key (pknull)
-		/*
-		sdo_writeUInt(sdow, 0);
-		sdo_writeUInt(sdow, 0);
-		sdow_begin_sequence(sdow);
-		sdo_writeUInt(sdow, 0);
-		sdow_end_sequence(sdow);
-		sdow_end_sequence(sdow);
-		*/
-		return;
-	}
-	// LOG(LOG_ERROR, "------- pk is %lu bytes long\n",
-	// pk->key1->byte_sz);
 	sdow_start_array(sdow, 3);
-	sdow_unsigned_int(sdow, pk->pkalg);
-	sdow_unsigned_int(sdow, pk->pkenc);
+	sdow_signed_int(sdow, pk->pkalg);
```

Comment:
Most of the sdow_xxx() calls either return an int or a bool. The return code needs to be handled appropriately.

Reply:
Missed it out earlier. Updated.